### PR TITLE
Update Pydantic to v2 and redis-om to v0.3.

### LINKED ISSUE #72: [FEAT]: Pydantic V2 + RedisOm Python V0.3
### Description

As https://github.com/redis/redis-om-python/pull/603 was merged three weeks ago, we can now support Pydantic V2 fully. 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ProKil**
Closes #72  

## 📑 Description

Update Pydantic to v2 and redis-om to v0.3. This PR is meant to be for v0.2 of sotopia, so please merge this after tagging v0.1.

~~Working, but still WIP: there are a few warnings to resolve~~

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed
- [x] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [x] Ready for code review

### CODE REVIEW

File: sotopia/database/logs.py [hunk 1]
```diff
@@ -14,29 +19,21 @@ class EpisodeLog(JsonModel):
 
     environment: str = Field(index=True)
     agents: list[str] = Field(index=True)
-    tag: str | None = Field(index=True)
-    models: list[str] | None = Field(index=True)
+    tag: str | None = Field(index=True, default="")
+    models: list[str] | None = Field(index=True, default=[])
```
Comment (bugsz) [inline sotopia/database/logs.py:23]: if the default is empty then is there any case it could be `None`?

Comment (ProKil) [reply sotopia/database/logs.py:23]: I kept that for compatibility reasons.

Comment (ProKil) [reply sotopia/database/logs.py:23]: I am not sure if the saved data before are none or ""/[]

File: sotopia/database/serialization.py [hunk 2]
```diff
@@ -14,66 +15,66 @@
...
-    raw_rewards: list[tuple[float, dict[str, float]] | float] = Field(required=True)
-    raw_rewards_prompt: str = Field(required=True)
-    scenario: str = Field(required=True)
-    codename: str = Field(required=True)
-    agents_background: dict[str, str] = Field(required=True)
-    social_goals: dict[str, str] = Field(required=True)
-    social_interactions: str = Field(required=True)
-    reasoning: str = Field(required=False)
-    rewards: list[tuple[float, dict[str, float]]] = Field(required=False)
+    episode_id: str = Field()
```
Comment (XuhuiZhou) [inline sotopia/database/serialization.py:18]: what's the default requirement now? is it still required=True.

File: sotopia/generation_utils/generate.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
 from langchain.schema import BaseOutputParser, OutputParserException
 from langchain_openai import ChatOpenAI, AzureChatOpenAI
 from pydantic import BaseModel, Field
+from pydantic.v1 import SecretStr
```
Comment (XuhuiZhou) [inline sotopia/generation_utils/generate.py:23]: why are we using this?

**Review by XuhuiZhou — APPROVED** [review]
Overall, I think this is good, but should we also update the doc at the same time?

File: sotopia/database/serialization.py (hunk 2 above)
Comment (ProKil) [reply sotopia/database/serialization.py:18]: Check out here: https://docs.pydantic.dev/latest/migration/#required-optional-and-nullable-fields. I think I maintained all of the semantics

File: sotopia/generation_utils/generate.py (hunk 3 above)
Comment (ProKil) [reply sotopia/generation_utils/generate.py:23]: required by langchain (sadly)

**Comment by ProKil** [comment]
Nothing from the usage perspective has changed. This can be verified by the fact that it passed all of the test cases for sotopia v0.1

**Pull request merged**